packages/component-base/src/url-utils.d.ts

```diff
@@ -0,0 +1,10 @@
+/**
+ * @license
+ * Copyright (c) 2023 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+
+/**
+ * Check if two paths match
+ */
+export declare function arePathsMatching(path1: string, path2: string): boolean;
```

Comment:
I would name this function `comparePaths(path1, path2)` - see for example this one in `vaadin-grid`:

https://github.com/vaadin/web-components/blob/ed4206da6fe731b75a0ba4ae67bdea6fdea0ed83/packages/grid/src/array-data-provider.js#L59

Reply:
Wouldn't that create an expectation that the helper returns a number?

Reply:
Right, my bad. Let's use `matchPaths` as it's short and self-explanatory.

Reply:
Done.